Utilities/src/main/java/edu/stanford/hivdb/mutations/MutationSet.java

```diff
@@ -240,22 +240,35 @@ public MutationSet mergesWith(Mutation... mutations) {
 	 */
 	public MutationSet intersectsWith(Collection<Mutation> another) {
 		Set<GenePosition> gpKeys = new HashSet<>(genePositionMap.keySet());
-		MutationSet anotherSet;
+		Set<GenePosition> gpKeysAnother;
 		if (another instanceof MutationSet) {
-			anotherSet = (MutationSet) another;
-		}
-		else {
-			anotherSet = new MutationSet(another);
+			gpKeysAnother = ((MutationSet) another).genePositionMap.keySet();
+		} else {
+			gpKeysAnother = another
+				.stream()
+				.map(m -> m.getGenePosition())
+				.collect(Collectors.toSet());
 		}
-		gpKeys.retainAll(anotherSet.genePositionMap.keySet());
-		List<Mutation> resultMuts = new ArrayList<>();
+		gpKeys.retainAll(gpKeysAnother);
+		List<Mutation> mutations = new ArrayList<>();
 		for (GenePosition gp : gpKeys) {
-			Set<Mutation> thisMuts = this.getSplitted(gp.gene, gp.position);
-			Set<Mutation> otherMuts = anotherSet.getSplitted(gp.gene, gp.position);
-			thisMuts.retainAll(otherMuts);
-			resultMuts.addAll(thisMuts);
+			Mutation thisMut = genePositionMap.get(gp);
+			Set<Character> otherAAChars = new TreeSet<>();
+			if (another instanceof MutationSet) {
```

Comment:
To avoid checking whether `another instanceof MutationSet` twice, I recommend converting `another` into a`MutationSet` near the top of this method and then proceeding with that guarantee.